docs/RFCS/system_jobs.md

```diff
@@ -0,0 +1,424 @@
+- Feature Name: System Jobs
+- Status: draft
+- Start Date: 2016-02-13
+- Authors: Nikhil Benesch
+- RFC PR: (PR # after acceptance of initial draft)
+- Cockroach Issue: [#12555]
+
+# Summary
+
+Add a system table to track the progress of backups, restores, and schema
+changes.
+
+# Motivation
+
+When performing a schema change, only one bit of progress information is
+available: whether the change has completed, indicated by whether the query has
+returned. Similarly, when performing a backup or restore, status is only
+reported after the backup or restore has completed. Given that a full backup of
+a 2TB database takes on the order of several hours to complete, the lack of
+progress information is a serious pain point for users.
+
+Additionally, if the leader of a backup or restore job fails, the backup or
+restore will abort, even if the individual workers in the job were otherwise
+successful.
+
+This RFC proposes a new system table, `system.jobs`, that tracks the status of
+these long-running backup, restore, and schema change "jobs." This table will
+directly expose the desired progress information via `SELECT` queries over the
+table, enable an API endpoint to expose the same progress information in the
+admin UI, and enable an internal daemon that periodically adopts and resumes
+orphaned jobs.
+
+# Detailed design
+
+## Prior art
+
+Adding a `system.jobs` table has been proposed and unsuccessfully implemented
+several times. [@vivekmenezes]'s initial attempt in [#7037] was abandoned since,
+at the time, there was no way to safely add a new system table to an existing
+cluster. Several months later, after [@a-robinson] and [@danhhz] built the
+necessary cluster upgrade framework, @a-robinson submitted a migration to add a
+`system.jobs` table in [#11722] as an example of using the new framework. His PR
+was rejected because the table's schema hadn't been fully considered. This RFC
+describes a revised `system.jobs` schema for us to thoroughly vet before we
+proceed with a final implementation.
+
+## The proposal
+
+Within this RFC, the term "job" or "long-running job" refers only to backups,
+restores, and schema changes. Other types of long-running queries, like slow
+`SELECT` statements, are explicitly out of scope. A "job-creating query," then,
+is any query of one of the following types:
+
+- `ALTER TABLE`, which creates a schema change job
+- `BACKUP`, which creates a backup job
+- `RESTORE`, which creates a restore job
+
+To track the progress of these jobs, the following table will be injected into
+the `system` database using the [cluster upgrade framework]:
+
+```sql
+CREATE TABLE system.jobs (
+    id               INT DEFAULT unique_rowid() PRIMARY KEY,
```

Comment:
I think you'll usually have a status if you're clicking through to an individual job from the admin ui. A secondary index on a system table just really feels like overkill to me. Fairly strong preference for updating this to one index and proposing the secondary index version in the alternatives section